src/test/java/io/pravega/connectors/hadoop/PravegaInputFormatTest.java

```diff
@@ -10,67 +10,109 @@
 public class PravegaInputFormatTest {
 
     private static final String TEST_SCOPE = "PravegaInputFormatTest";
     private static final String TEST_STREAM = "stream";
-    private static final int NUM_SEGMENTS = 3;
-    private static final int NUM_EVENTS = 20;
-
-    /**
-     * Setup utility
-     */
-    private static final SetupUtils SETUP_UTILS = new SetupUtils();
+    private static final String TEST_URI = "tcp://127.0.0.1:9090";
 
     @Before
-    public void setupPravega() throws Exception {
-        SETUP_UTILS.startAllServices(TEST_SCOPE);
-        SETUP_UTILS.createTestStream(TEST_STREAM, NUM_SEGMENTS);
-        EventStreamWriter<Integer> writer = SETUP_UTILS.getIntegerWriter(TEST_STREAM);
-        for (int i = 0; i < NUM_EVENTS; i++) {
-            CompletableFuture future = writer.writeEvent(i);
-            future.get();
-        }
+    public void setupTests() throws Exception {
+        mockClientFactoryStatic(TEST_SCOPE, TEST_URI);
     }
 
-    @After
-    public void tearDownPravega() throws Exception {
-        SETUP_UTILS.stopAllServices();
+    @Test
+    public void testGetSplits() throws Exception {
+        JobContext ctx = getJobContext();
+        Configuration c = ctx.getConfiguration();
+        Assert.assertNotNull(c);
+        PravegaInputFormat<Integer> inputFormat = new PravegaInputFormat();
+        List<InputSplit> splits = inputFormat.getSplits(ctx);
+        Assert.assertEquals(splits.size(), 3);
+        int i = 0;
+        for (InputSplit s : splits) {
+            Assert.assertTrue(s instanceof PravegaInputSplit);
+            PravegaInputSplit ps = (PravegaInputSplit) s;
+            i++;
+            Assert.assertTrue(0 == ps.compareTo(new PravegaInputSplit(new Segment(TEST_SCOPE, TEST_STREAM, i), 0, 100 * i)));
+        }
     }
 
     @Test
-    public void testGetSplits() throws IOException, InterruptedException {
+    public void testCreateRecordReader() throws IOException, InterruptedException {
+        PravegaInputFormat<Integer> inputFormat = new PravegaInputFormat();
+        RecordReader<?, ?> reader = inputFormat.createRecordReader(null, null);
+        Assert.assertTrue(reader instanceof PravegaInputRecordReader);
+    }
+
+    private JobContext getJobContext() throws Exception {
         Configuration conf = new Configuration();
         conf.setStrings(PravegaInputFormat.SCOPE_NAME, TEST_SCOPE);
         conf.setStrings(PravegaInputFormat.STREAM_NAME, TEST_STREAM);
-        conf.setStrings(PravegaInputFormat.URI_STRING, SETUP_UTILS.getControllerUri());
+        conf.setStrings(PravegaInputFormat.URI_STRING, TEST_URI);
         conf.setStrings(PravegaInputFormat.DESERIALIZER, IntegerSerializer.class.getName());
-        Job job = new Job(conf);
+        Job mockJob = mock(Job.class);
+        PowerMockito.whenNew(Job.class).withAnyArguments().thenReturn(mockJob);
+        Mockito.doReturn(conf).when(mockJob).getConfiguration();
+        return new Job(conf);
```

Comment:
I don't understand why powermock is needed in this test case.   Why `return new Job()` instead of `return mockJob`?

Reply:
good catch. accepted